modules/examples/todolist/src/main/scala/todo/http/apis/TagApi.scala

```diff
@@ -18,56 +18,61 @@ package todo
 package http
 package apis
 
-import cats.~>
-import cats.instances.list._
-import cats.syntax.traverse._
+import cats.{~>, Monad}
+import cats.Monad.ops._
 import com.twitter.util.Future
 import io.finch._
 import io.finch.circe._
 import io.circe.generic.auto._
-import freestyle.free._
-import freestyle.free.http.finch._
-import freestyle.free.logging._
-import todo.model.Tag
-import todo.services._
-import todo.model.Tag
+import examples.todolist.Tag
+import examples.todolist.service.TagService
 
-class TagApi[F[_]](implicit service: TagService[F], handler: F ~> Future) extends CRUDApi[Tag] {
+class TagApi[F[_]: Monad](implicit service: TagService[F], handler: F ~> Future)
+    extends CRUDApi[Tag] {
 
-  import io.finch.syntax._
+  import io.finch.syntax.post
+  import io.finch.syntax.get
+  import io.finch.syntax.put
+  import io.finch.syntax.delete
 
-  val reset = post(service.prefix :: "reset") {
-    service.reset.map(Ok)
+  private val prefix = "/tags"
+  private val model  = classOf[Tag].getSimpleName
```

Comment:
Is there is any other alternative to extract the model name?

Reply:
As in the original example, I've changed it so it takes the model from the `service`.